### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'groupdict'

### ISSUE THREAD

**Issue by macmanes** [camillescott/dammit#96]
Installed dammit in a virtual env like described here: https://angus.readthedocs.io/en/2017/dammit_annotation.html

Any hints about the error?

```
(py3) [macmanes@premise frog]$ dammit annotate Highexp5.fasta --busco-group tetrapoda --database-dir $SHARE/dammit_dbs --n_threads 24 --name Rimi --full
# dammit
## a tool for easy de novo transcriptome annotation

by Camille Scott

**v1.0.dev0**, 2017

## submodule: annotate
### Database Check
#### Info
* Database Directory: /mnt/lustre/macmaneslab/shared/dammit_dbs
* Doit Database: /mnt/lustre/macmaneslab/shared/dammit_dbs/databases.doit.db


*All database tasks up-to-date.*

### Annotation
#### Info
* Doit Database: /mnt/lustre/macmaneslab/macmanes/frog/Highexp5.fasta.dammit/annotate.doit.db
* Input Transcriptome: /mnt/lustre/macmaneslab/macmanes/frog/Highexp5.fasta.dammit/Highexp5.fasta


Some tasks out of date!


Up-to-date tasks:
* TransDecoder.LongOrfs
* hmmscan:Pfam-A
* rename-transcriptome
* transcriptome-stats


Out-of-date tasks:
* BUSCO-tetrapoda
* TransDecoder.Predict
* annotate:fasta
* cmscan:Rfam
* gff3:OrthoDB
* gff3:Pfam-A
* gff3:Rfam
* gff3:merge-all
* gff3:uniref90
* hmmscan:Pfam-A:remap
* lastal:OrthoDB
* lastal:best-hits:OrthoDB
* lastal:best-hits:uniref90
* lastal:uniref90

#### Run Tasks
- [x] Highexp5.fasta:
    * Python: function get_rename_transcriptome_task.fix
- [x] transcriptome_stats:Highexp5.fasta:
    * Python: function get_transcriptome_stats_task.cmd
- [ ] busco:Highexp5.fasta-tetrapoda_odb9:
    * Cmd: `BUSCO_CONFIG_FILE="/mnt/lustre/macmaneslab/macmanes/py3/lib/python3.6/site-packages/dammit/busco_config.ini" python3 /mnt/lustre/macmaneslab/shared/software/busco/scripts/run_BUSCO.py -i /mnt/lustre/macmaneslab/macmanes/frog/Highexp5.fasta.dammit/Highexp5.fasta -f -o Highexp5.fasta.tetrapoda.busco.results -l /mnt/lustre/macmaneslab/shared/dammit_dbs/busco2db/tetrapoda_odb9 -m tran -c 24`
- [x] TransDecoder.LongOrfs:Highexp5.fasta:
    * Cmd: `/mnt/lustre/software/linuxbrew/colsa/bin/TransDecoder.LongOrfs -t /mnt/lustre/macmaneslab/macmanes/frog/Highexp5.fasta.dammit/Highexp5.fasta -m 80`
- [x] hmmscan:longest_orfs.pep.x.Pfam-A.hmm:
    * Cmd: `cat /mnt/lustre/macmaneslab/macmanes/frog/Highexp5.fasta.dammit/Highexp5.fasta.transdecoder_dir/longest_orfs.pep | /mnt/lustre/software/linuxbrew/colsa/bin/parallel --round-robin --pipe -L 2 -N 10000 --gnu -j 24 -a /mnt/lustre/macmaneslab/macmanes/frog/Highexp5.fasta.dammit/Highexp5.fasta.transdecoder_dir/longest_orfs.pep /mnt/lustre/software/linuxbrew/colsa/bin/hmmscan --cpu 1 --domtblout /dev/stdout -E 1e-05 -o /mnt/lustre/macmaneslab/macmanes/frog/Highexp5.fasta.dammit/Highexp5.fasta.transdecoder_dir/longest_orfs.pep.x.pfam.tbl.hmmscan.out /mnt/lustre/macmaneslab/shared/dammit_dbs/Pfam-A.hmm /dev/stdin > /mnt/lustre/macmaneslab/macmanes/frog/Highexp5.fasta.dammit/Highexp5.fasta.transdecoder_dir/longest_orfs.pep.x.pfam.tbl`
- [ ] remap_hmmer:longest_orfs.pep.x.pfam.tbl:
    * Python: function get_remap_hmmer_task.cmd
########################################
TaskError - taskid:remap_hmmer:longest_orfs.pep.x.pfam.tbl
PythonAction Error
Traceback (most recent call last):
  File "/mnt/lustre/macmaneslab/macmanes/py3/lib/python3.6/site-packages/doit/action.py", line 403, in execute
    returned_value = self.py_callable(*self.args, **kwargs)
  File "/mnt/lustre/macmaneslab/macmanes/py3/lib/python3.6/site-packages/dammit/tasks/hmmer.py", line 137, in cmd
    hmmer_df = HMMerParser(hmmer_filename).read()
  File "/mnt/lustre/macmaneslab/macmanes/py3/lib/python3.6/site-packages/dammit/fileio/base.py", line 73, in read
    return pd.concat(self, ignore_index=True)
  File "/mnt/lustre/macmaneslab/macmanes/py3/lib/python3.6/site-packages/pandas/core/reshape/concat.py", line 206, in concat
    copy=copy)
  File "/mnt/lustre/macmaneslab/macmanes/py3/lib/python3.6/site-packages/pandas/core/reshape/concat.py", line 236, in __init__
    objs = list(objs)
  File "/mnt/lustre/macmaneslab/macmanes/py3/lib/python3.6/site-packages/dammit/fileio/hmmer.py", line 67, in __iter__
    yield self._build_df(data)
  File "/mnt/lustre/macmaneslab/macmanes/py3/lib/python3.6/site-packages/dammit/fileio/hmmer.py", line 91, in _build_df
    df['query_name'] = df.query_name.apply(split_query)
  File "/mnt/lustre/macmaneslab/macmanes/py3/lib/python3.6/site-packages/pandas/core/series.py", line 2355, in apply
    mapped = lib.map_infer(values, f, convert=convert_dtype)
  File "pandas/_libs/src/inference.pyx", line 1574, in pandas._libs.lib.map_infer (pandas/_libs/lib.c:66645)
  File "/mnt/lustre/macmaneslab/macmanes/py3/lib/python3.6/site-packages/dammit/fileio/hmmer.py", line 80, in split_query
    results = self.query_regex.search(item).groupdict()
AttributeError: 'NoneType' object has no attribute 'groupdict'


/mnt/lustre/macmaneslab/macmanes/py3/lib/python3.6/site-packages/dammit/fileio/gff3.py:68: ParserWarning: Both a converter and dtype were specified for column attributes - only the converter will be used
  dtype=dict(self.columns)):

```

**Comment by macmanes** [comment]
Hey @camillescott any update on this? cc: @ctb

**Comment by macmanes** [comment]
of note, `--quick` works, but dammit fails during `function get_remap_hmmer_task.cmd` with the full run.

**Comment by ljcohen** [comment]
replicated this with fresh install from [instructions](http://dibsi-rnaseq.readthedocs.io/en/latest/dammit_annotation.html), with this command with files from [tutorial](http://dibsi-rnaseq.readthedocs.io/en/latest/dammit_annotation.html):

```

dammit annotate trinity.nema.fasta --busco-group metazoa --user-databases nema.reference.prot.faa --n_threads 6
```

```
(py3) ljcohen@js-169-43:~$ dammit annotate trinity.nema.fasta --busco-group metazoa --user-databases nema.reference.prot.faa --n_threads 6
# dammit
## a tool for easy de novo transcriptome annotation

by Camille Scott

**v1.0.dev0**, 2017

## submodule: annotate
### Database Check
#### Info
* Database Directory: /vol_b/dammit/
* Doit Database: /vol_b/dammit/databases.doit.db


*All database tasks up-to-date.*

### Annotation
#### Info
* Doit Database: /home/ljcohen/trinity.nema.fasta.dammit/annotate.doit.db
* Input Transcriptome: /home/ljcohen/trinity.nema.fasta.dammit/trinity.nema.fasta


Some tasks out of date!


Up-to-date tasks:
* BUSCO-metazoa
* TransDecoder.LongOrfs
* hmmscan:Pfam-A
* rename-transcriptome
* transcriptome-stats


Out-of-date tasks:
* TransDecoder.Predict
* annotate:fasta
* cmscan:Rfam
* gff3:/home/ljcohen/trinity.nema.fasta.dammit/trinity.nema.fasta.x.nema.reference.prot.faa.crbl.csv
* gff3:OrthoDB
* gff3:Pfam-A
* gff3:Rfam
* gff3:merge-all
* hmmscan:Pfam-A:remap
* lastal:OrthoDB
* lastal:best-hits:OrthoDB
* user-database:nema.reference.prot.faa-shmlast-fit_and_filter_crbl_hits
* user-database:nema.reference.prot.faa-shmlast-lastal:.nema.reference.prot.faa.x.trinity.nema.fasta.pep.maf
* user-database:nema.reference.prot.faa-shmlast-lastal:.trinity.nema.fasta.pep.x.nema.reference.prot.faa.maf
* user-database:nema.reference.prot.faa-shmlast-lastdb:.nema.reference.prot.faa
* user-database:nema.reference.prot.faa-shmlast-lastdb:.trinity.nema.fasta.pep
* user-database:nema.reference.prot.faa-shmlast-rename:/home/ljcohen/nema.reference.prot.faa
* user-database:nema.reference.prot.faa-shmlast-rename:/home/ljcohen/trinity.nema.fasta.dammit/trinity.nema.fasta
* user-database:nema.reference.prot.faa-shmlast-translate:.trinity.nema.fasta

#### Run Tasks
- [x] trinity.nema.fasta: 
    * Python: function get_rename_transcriptome_task.fix
- [x] transcriptome_stats:trinity.nema.fasta: 
    * Python: function get_transcriptome_stats_task.cmd
- [x] busco:trinity.nema.fasta-metazoa_odb9: 
    * Cmd: `BUSCO_CONFIG_FILE="/home/ljcohen/py3/lib/python3.5/site-packages/dammit/busco_config.ini" python3 /home/ljcohen/busco/scripts/run_BUSCO.py -i /home/ljcohen/trinity.nema.fasta.dammit/trinity.nema.fasta -f -o trinity.nema.fasta.metazoa.busco.results -l /vol_b/dammit/busco2db/metazoa_odb9 -m tran -c 6`
- [x] TransDecoder.LongOrfs:trinity.nema.fasta: 
    * Cmd: `/usr/bin/TransDecoder.LongOrfs -t /home/ljcohen/trinity.nema.fasta.dammit/trinity.nema.fasta -m 80`
- [x] hmmscan:longest_orfs.pep.x.Pfam-A.hmm: 
    * Cmd: `cat /home/ljcohen/trinity.nema.fasta.dammit/trinity.nema.fasta.transdecoder_dir/longest_orfs.pep | /home/ljcohen/bin/parallel --round-robin --pipe -L 2 -N 10000 --gnu -j 6 -a /home/ljcohen/trinity.nema.fasta.dammit/trinity.nema.fasta.transdecoder_dir/longest_orfs.pep /usr/bin/hmmscan --cpu 1 --domtblout /dev/stdout -E 1e-05 -o /home/ljcohen/trinity.nema.fasta.dammit/trinity.nema.fasta.transdecoder_dir/longest_orfs.pep.x.pfam.tbl.hmmscan.out /vol_b/dammit/Pfam-A.hmm /dev/stdin > /home/ljcohen/trinity.nema.fasta.dammit/trinity.nema.fasta.transdecoder_dir/longest_orfs.pep.x.pfam.tbl`
- [ ] remap_hmmer:longest_orfs.pep.x.pfam.tbl: 
    * Python: function get_remap_hmmer_task.cmd
########################################
TaskError - taskid:remap_hmmer:longest_orfs.pep.x.pfam.tbl
PythonAction Error
Traceback (most recent call last):
  File "/home/ljcohen/py3/lib/python3.5/site-packages/doit/action.py", line 403, in execute
    returned_value = self.py_callable(*self.args, **kwargs)
  File "/home/ljcohen/py3/lib/python3.5/site-packages/dammit/tasks/hmmer.py", line 137, in cmd
    hmmer_df = HMMerParser(hmmer_filename).read()
  File "/home/ljcohen/py3/lib/python3.5/site-packages/dammit/fileio/base.py", line 73, in read
    return pd.concat(self, ignore_index=True)
  File "/home/ljcohen/py3/lib/python3.5/site-packages/pandas/tools/merge.py", line 845, in concat
    copy=copy)
  File "/home/ljcohen/py3/lib/python3.5/site-packages/pandas/tools/merge.py", line 875, in __init__
    objs = list(objs)
  File "/home/ljcohen/py3/lib/python3.5/site-packages/dammit/fileio/hmmer.py", line 73, in __iter__
    yield self._build_df(data)
  File "/home/ljcohen/py3/lib/python3.5/site-packages/dammit/fileio/hmmer.py", line 91, in _build_df
    df['query_name'] = df.query_name.apply(split_query)
  File "/home/ljcohen/py3/lib/python3.5/site-packages/pandas/core/series.py", line 2220, in apply
    mapped = lib.map_infer(values, f, convert=convert_dtype)
  File "pandas/src/inference.pyx", line 1088, in pandas.lib.map_infer (pandas/lib.c:62658)
  File "/home/ljcohen/py3/lib/python3.5/site-packages/dammit/fileio/hmmer.py", line 80, in split_query
    results = self.query_regex.search(item).groupdict()
AttributeError: 'NoneType' object has no attribute 'groupdict'
```

Thought might have been issue with new version of [pandas 0.20.3](http://pandas.pydata.org/) so rolled back to 0.19.2 with:

```
pip uninstall pandas
pip install -Iv pandas==0.19.2
```
but same error. Could be input is empty for some reason? Found this:

https://stackoverflow.com/questions/30797220/python-3-x-attributeerror-nonetype-object-has-no-attribute-groupdict

**Comment by camillescott** [comment]
Pinned it down to an issue with how `--name` is being (or not being) handled. Fix incoming.

**Comment by camillescott** [comment]
This has now been fixed in the `refactor/1.0` branch.
